### PR TITLE
EC-2681 Added age consent fields and validation

### PR DESCRIPTION
**Pull request by andyjmaclean**

### CODE REVIEW

File: app/models/story.rb [hunk 1]
```diff
@@ -15,10 +15,16 @@ class Story
   index(created_at: 1)
   index(updated_at: 1)
 
+  field :age_confirm
```
Comment (rwd) [inline app/models/story.rb]: These need to be typed, I expect as Boolean.

File: app/models/story.rb [hunk 2]
```diff
@@ -15,10 +15,16 @@ class Story
...
+  field :guardian_consent
+
   accepts_nested_attributes_for :ore_aggregation
 
   validates_associated :ore_aggregation
 
+  validates :age_confirm, acceptance: { accept: ['true', 1] } unless :guardian_consent
```
Comment (rwd) [inline app/models/story.rb]: These should be showing validation error messages if not accepted, even without JS validation.

File: config/locales/migration.en.yml [hunk 3]
```diff
@@ -34,6 +34,9 @@ en:
               rdaGr2_dateOfDeath: Date of death
               rdaGr2_placeOfBirth: Place of birth
               rdaGr2_placeOfDeath: Place of death
+            story:
+              age_confirm: I am over 16 years old
+              guardian_consent: I am under 16 years of age and my parents have expressed doumented consent that I can submit this form
```
Comment (rwd) [inline config/locales/migration.en.yml]: This needs spelling correction, at least.

File: spec/system/story_submittal_and_retrieval_spec.rb [hunk 4]
```diff
@@ -7,21 +7,25 @@
 require 'sidekiq/api'
 
 RSpec.describe 'story submittal and retrieval', sidekiq: true do
+  before do
+    # TODO: When form saving is fully functional consider enabling it here
```
Comment (rwd) [inline spec/system/story_submittal_and_retrieval_spec.rb]: Why is this here? It's a feature toggle which is off by default.

File: app/models/story.rb [hunk 5]
```diff
@@ -15,10 +15,16 @@ class Story
...
+  validates :guardian_consent, acceptance: { accept: ['true', 1] } unless :age_confirm
```
Comment (rwd) [inline app/models/story.rb]: As a non-authenticated user, I can't ever get the form to save now, but no error message is shown, with JS & reCAPTCHA disabled.

File: config/locales/migration.en.yml (hunk 3 above)
Comment (lbiedinger) [reply config/locales/migration.en.yml]: moved to localeapp and corrected there

File: spec/system/story_submittal_and_retrieval_spec.rb (hunk 4 above)
Comment (lbiedinger) [reply spec/system/story_submittal_and_retrieval_spec.rb]: the tests for for asseted_views_spec were setting env variables, so depending on what was ran before here and while the form rembemering isn't working this is needed for consistency.

File: app/models/story.rb [hunk 6]
```diff
@@ -19,10 +19,16 @@ class Story
...
   index(aasm_state: 1)
 
+  field :age_confirm, type: Boolean, default: false
+  field :guardian_consent, type: Boolean, default: false
+
   accepts_nested_attributes_for :ore_aggregation
 
   validates_associated :ore_aggregation
 
+  validates :age_confirm, acceptance: { accept: [true, 1], message: I18n.t('global.forms.validation-errors.user-age') }, unless: :guardian_consent
```
Comment (rwd) [inline app/models/story.rb]: Could we have a validation sanity check here preventing both age_confirm and guardian_consent being accepted on one story? i.e. if a contributor is over 16, then they can not also be under 16.

File: app/models/story.rb [hunk 7]
```diff
@@ -95,4 +105,12 @@ def sets
   def has_media?
     ore_aggregation.edm_web_resources.any?(&:media?)
   end
+
+  def guardian_consent?
```
Comment (rwd) [inline app/models/story.rb]: `#guardian_consent?` and `#age_confirm?` are automatically added to the model by Mongoid and need not be declared explicitly.
